### PR TITLE
PEExperiment: 1. Copy/Paste properties of the same class. 2. Copy a BMO or MIC mat into the opposite. 3. Remove references to SM.

### PR DESCRIPTION
**Pull request by Exkywor**
This branch has three experiments:
1. Experiment to copy any property from a selected export into another export of the same class.
2. Experiment to copy the material properties of a BioMaterialOverrides or MaterialInstanceConstant into an export of the opposite class.
3. Experiment to remove the [Static/Skeletal]MeshComponent references to a [Static/Skeletal]Mesh that are within a given distance from a given origin.

### CODE REVIEW

File: LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs [hunk 1]
```diff
@@ -490,5 +551,216 @@ class PackageEditorExperimentsO {
...
+                scalarProperty.ToList().ForEach(parameter =>
+                {
+                    PropertyCollection props = new();
+                    if (isBMO) { props.Add(GenerateExpressionGUID()); }
+                    props.Add(new NameProperty(parameter.GetProp<NameProperty>($"{(isBMO ? "nName" : "ParameterName")}").Value,
+                        $"{(isBMO ? "ParameterName" : "nName")}"));
+                    props.Add(new FloatProperty(parameter.GetProp<FloatProperty>($"{(isBMO ? "sValue" : "ParameterValue")}").Value,
+                        $"{(isBMO ? "ParameterValue" : "sValue")}"));
+                    ScalarValues.Add(new StructProperty($"{(isBMO ? "ScalarParameterValue" : "ScalarParameter")}", props));
+                });
```
Comment (SirCxyrtyx) [inline LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs]: This should be a regular foreach loop. There's no point in converting it into a list and then calling a ForEach function on it. Same goes for the other two places you've done this.

**Comment by SirCxyrtyx** [comment]
This PR has some merge conflicts now, due to merge of your other PR. Could you merge beta into your branch to resolve those?

File: LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs (hunk 1 above)
Comment (Exkywor) [reply LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs]: Got it

**Comment by Exkywor** [comment]
> This PR has some merge conflicts now, due to merge of your other PR. Could you merge beta into your branch to resolve those?

Sure thing. Do you want me to also merge this with my other experiment as to make a single pull request? Mg told me it was fine having them separate, but if it's easier for you I can merge them on my end

**Comment by SirCxyrtyx** [comment]
Yeah. merging it with the other pr would be good, else we'll just have to do the same thing again, since its all edits to the same file. Hadn't anticipated the merge conflict problem.

**Comment by Exkywor** [comment]
Alright, I'll make the changes and merge and push to this branch

**Comment by Exkywor** [comment]
Merges done

**Comment by SirCxyrtyx** [comment]
It's still showing conflicts. Not sure why. Might be simpler at this point to just make a new branch based off of beta and copy your changes into it.

**Comment by Exkywor** [comment]
Got it. I'll delete this pull request and make a new one then

**Pull request closed without merging**